### DISCUSSION TITLE
How to use cron jobs with Vercel?

### DISCUSSION THREAD

**Discussion by bitsmyth** [vercel/vercel#5344] [Help] [answered]
I would like to call a serverless next.js API in regular intervals without using third party services.
How should I approach this?

**Comment by mcsdevv** [comment]
Hello @NikoMontana, if you want to do this from the client you can use [SWR](https://swr.vercel.app/). However, if you are looking to handle this from the server-side, you will need to use a third-party service, this is because Serverless Functions are stateless.

**Comment by leerob** [comment, marked as answer]
**Edit: Update 2023**

Wanted to share an exciting update – we just released Vercel Cron Jobs!

Vercel Cron Jobs enable you to run scheduled jobs for automating backups and archiving, sending email and Slack notifications, and more. Cron Jobs can be used for any task you need to run on a schedule.

https://vercel.com/blog/cron-jobs

**Reply by fabioluiz1** [reply]
Regarding security https://vercel.com/docs/cron-jobs#how-to-secure-cron-jobs

Is there a way to use environment variables to make sure the cron api endpoint was initiated by a Vercel Cron scheduler?

**Reply by Peek-A-Booo** [reply]
Is there any suggestions for cron security?
According to the official documentation, we can use the query string of a path to share a key with your function. However, if the repository is open source, submitting this key to the code repository will be of no use.

**Reply by mcsdevv** [reply]
Hey @Peek-A-Booo this is possible using an env var rather than hard coding, see [here](https://github.com/orgs/vercel/discussions/1734#discussioncomment-6146073).

We'll get this documented soon!

**Reply by Peek-A-Booo** [reply]
thanks, it works for me🥳
